### DISCUSSION TITLE
Add support for bulk data

### DISCUSSION THREAD

**Discussion by alederzz** [ash-jc-allen/short-url#160] [Ideas]
Would it be possible to add a bulk upload method?

I noticed that in the individual creation a query is generated to obtain the last ID, +1 is added, the hash is generated and it is checked again if it already exists. If it exists, the process is repeated.

https://github.com/ash-jc-allen/short-url/blob/master/src/Classes/KeyGenerator.php#L40

```
public function generateRandom(): string
    {
        $ID = $this->getLastInsertedID();

        do {
            $ID++;
            $key = $this->hashids->encode($ID);
        } while (ShortURL::where('url_key', $key)->exists());

        return $key;
    }
```

But this process creates problems if we want to run multiple jobs in parallel.

Probably a case that I still do not have very clear would be to insert the destinations in a massive way and with another method generate the hashes.

```
// destination , url_key
$data= [
     ['https://five.xyz','KeyFive'],
       ...
     ['https://monday.xyz', null],
     ['https://orange.xyz','oRanGe'],
];

$destinations = (new Builder())->create($data); // only insert data

$destinations->each( function($destination) {
   // generate hash process
});

```

**Comment by ash-jc-allen** [comment]
Hey @alederzz! Thanks for the suggestion of the idea.

At the moment, I don't think I have any plans to add a feature like this, sorry. But if enough people want it though, it might be something that I'll consider adding.

For the time being, I think it might be best to build your URL keys yourself rather than allowing the package to build a random one for you. That way, you can choose the logic that you want to use for generating your URLs 🙂
